src/main/java/com/labsynch/labseer/domain/StandardizationDryRunCompound.java

```diff
@@ -909,5 +909,44 @@ public static Collection<StandardizationDryRunCompound> fromJsonArrayToStandardi
 				.use("values", StandardizationDryRunCompound.class).deserialize(json);
 	}
 
+	@Transactional
+	public static List<Long> fetchUnlockedParentIds(int limit) {
+		String sql = "SELECT p.id " +
+					"FROM parent p " +
+					"WHERE NOT EXISTS (" +
+					"    SELECT 1 " +
+					"    FROM standardization_dry_run_compound s " +
+					"    WHERE s.parent_id = p.id" +
+					") " +
+					"ORDER BY p.id ASC " +
+					"FOR UPDATE SKIP LOCKED " +
```

Comment:
This uses FOR UPDATE SKIP LOCKED query.   This allows us to checkout chunks of parent ids which have not yet been added to the standardization dry run compound without getting blocked by other processes.  If you run this query in 2 separate transactions, the first query will select e.g. ids 1-1000 and the next query will select ids 1001-2000.